lib/song.rb

```diff
@@ -0,0 +1,44 @@
+require "pry"
+
+class Song 
+  attr_accessor :name, :artist, :genre
+  
+  @@count = 0
+  @@artists = []
+  @@genres = []
+
+  def initialize(name, artist, genre)
+    @name = name
+    @artist = artist
+    @genre = genre
+
+    @@artists << artist
+    @@genres << genre
+    @@count += 1
+  end
+
+  def self.count
+    @@count
+  end
+
+  def self.artists
+    @@artists.filter.uniq do |artist|
+      artist
+    end
+  end
+
+  def self.genres
+    @@genres.filter.uniq do |genre|
+      genre
+    end
```

Comment:
No need to filter here. 
```suggestion
    @@genres.uniq
```